loadNeustarReputationIntoES.js

```diff
@@ -0,0 +1,1288 @@
+                            lastUsedIndex = lastUsedIndex.trim();
+                        }
+                    }
+
+                    if (program.deleteSingle == currentMonolithIndex) {
+                        winston.log('warn', "Failed to delete " + program.deleteSingle + " since it's currently being used by monolith.");
+                    } else if (program.deleteSingle == lastUsedIndex) {
+                        winston.log('warn', "Failed to delete " + program.deleteSingle + " since it was the last used index by monolith.");
+                    } else {
+                        winston.log('verbose', "About to delete " + program.deleteSingle);
+                        deleteIndex(program.deleteSingle, function (deleteSuccess, indexDeleted) {
+                            if (deleteSuccess) {
+                                if (indexDeleted === loadedIndex) {
+                                    removeLoadedIndexFromScratchSpace();
+                                }
+                                if (indexDeleted === pausedIndex) {
+                                    removePausedIndexFromScratchSpace();
+                                }
+                            }
+                        });
+                    }
+                });
+            });
+        } else {
+            throw new Error("Scratch space creation failed");
+        }
+    });
+}
+
+function getAllNeustarIndexes(cb) {
+    winston.log('verbose', "Getting all Neustar indexes");
+    //get all indexes that match regexp 'indexName'
+    var options = {
+        host: program.host,
+        port: program.port,
+        path: '/_cat/indices/' + indexRegExp + '?pri&v&h=index&pretty',
+        method: 'GET'
+    };
+
+    var callback = function(response) {
+        var str = '';
+        //another chunk of data has been recieved, so append it to `str`
+        response.on('data', function (chunk) {
+            str += chunk;
+        });
+
+        //the whole response has been recieved, so we just print it out here
+        response.on('end', function () {
+            winston.log('verbose', "Neustar indexes on ES are: " + str);
+
+            var arrayOfIndexes = str.split(/\r\n|\r|\n/);
+            var indexToDelete = null;
+            //we subtract 2 from teh length since there is a new line at the beginning of this list and at the end
+            //need to omit those 2 newlines when getting the real count of indexes
+            var indexCount = arrayOfIndexes.length - 2;
+            winston.log('verbose', "Number of indexes = " + indexCount);
+            if (indexCount === 0) {
+                cb(null);
+            } else {
+                //remove the first element since that just says "index" (header of the table)
+                arrayOfIndexes.shift();
+                //remove last line which is a blank line
+                arrayOfIndexes.pop();
+                for (var i = 0; i < indexCount; i++) {
```

Comment:
use `arrayOfIndexes = arrayOfIndexes.map(function(index) { return index.toString().trim() })` which you can then chain `.sort()` on